core/primitives/src/shard_layout.rs

```diff
@@ -241,9 +339,34 @@ pub enum ShardLayoutError {
 }
 
 impl ShardLayout {
-    /* Some constructors */
-    pub fn v0_single_shard() -> Self {
-        Self::v0(1, 0)
+    /// Handy constructor for a single-shard layout, mostly for test purposes
+    pub fn single_shard() -> Self {
+        Self::multi_shard(1, 0)
+    }
+
+    /// Can be used to construct a multi-shard layout, mostly for test purposes
+    pub fn multi_shard(num_shards: NumShards, version: ShardVersion) -> Self {
+        assert!(num_shards > 0, "at least 1 shard is required");
+
+        let boundary_accounts = (0..num_shards - 1)
+            .map(|i| format!("shard{}.test.near", i).parse().unwrap())
+            .collect::<Vec<AccountId>>();
+        let shard_ids = (0..num_shards).map(ShardId::new).collect::<Vec<ShardId>>();
+        let (id_to_index_map, index_to_id_map) = shard_ids
+            .iter()
+            .enumerate()
+            .map(|(i, &shard_id)| ((shard_id, i), (i, shard_id)))
+            .unzip();
+
+        Self::V2(ShardLayoutV2 {
+            boundary_accounts,
+            shard_ids,
+            id_to_index_map,
+            index_to_id_map,
+            shards_split_map: None,
+            shards_parent_map: None,
+            version,
+        })
     }
 
     /// Return a V0 Shardlayout
```

Comment:
Can you mark it as deprecated? I don't know how to do this properly in rust, if it's not straight forward then a comment should do.

Reply:
How about both? :)

Reply:
On second thought, probably just on `v0()` as there're still many occurrences of `V0` in the code which will probably cause clippy to fail on ci as it raises a deprecation warning (need something like `#[allow(deprecated)]` to suppress the warning).